src/mds/MDCache.cc

```diff
@@ -6512,7 +6512,8 @@ std::pair<bool, uint64_t> MDCache::trim_lru(uint64_t count, expiremap& expiremap
   unexpirables.clear();
 
   // trim dentries from the LRU until count is reached
-  while (!throttled && (cache_toofull() || count > 0)) {
+  // if mds is in standbyreplay and will trim all inodes which aren't in segments
```

Comment:
:O Why do we want to trim dentries not in a log segment? Wouldn't this prevent the standby-replay MDS from keeping its cache hot with all recently used metadata?

Reply:
standby replay mds only knows which metadata are recently modified.  recently modified metatdata is not equivalent to hot metadata